Sources/DualCameraKit/Screen/DualCameraViewModel.swift

```diff
@@ -190,9 +184,9 @@ final class DualCameraViewModel {
                 
                 // Reset recording state
                 viewState = .ready
-                
-                saveVideoToPhotoLibrary(videoRecordingOutputURL)
-                
+
+                // TODO: ensure to test permission denied
+                try await self.videoSaveStrategy.save(videoRecordingOutputURL)
             } catch let error as DualCameraError {
```

Comment:
update here: this error should not be 
`"showError(error, message: "Failed to stop recording)" ` 

it should be based on what the error is. e.g., now this error could be that there was a permission error